### PR TITLE
fix for searchEnabled option and corrects types

### PR DESCRIPTION
**Pull request by utrumo**
…of selects

## Description

From docs:
searchEnabled
Type: Boolean Default: true
Input types affected: select-one
Usage: Whether a search area should be shown.
Note: Multiple select boxes will always show search areas.

## Motivation and Context

https://github.com/jshjohnson/Choices/issues/533

## How Has This Been Tested?

https://jsfiddle.net/pvvnxdementor5/9gmv72L1/17/

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/scripts/choices.js [hunk 1]
```diff
@@ -117,6 +117,10 @@ class Choices {
       );
     }
 
+    this.config.searchEnabled = this._isSelectOneElement
+      ? this.config.searchEnabled
+      : true;
+
```
Comment (tinovyatkin) [inline src/scripts/choices.js]: I think this should be simplified to [documented logic](https://github.com/jshjohnson/Choices/blob/master/README.md#searchenabled):
```js
this.config.searchEnabled = this._isSelectMultipleElement || this.config.searchEnabled;
```

File: types/index.d.ts [hunk 2]
```diff
@@ -9,11 +9,13 @@ declare namespace Choices {
   namespace Types {
     type renderSelected = 'auto' | 'always';
     type dropdownPosition = 'auto' | 'top';
+    type passedElementType = 'select-one' | 'select-multiple' | 'text';
     type strToEl = (str: string) => HTMLElement | HTMLInputElement | HTMLOptionElement;
     type stringFunction = () => string;
     type noticeStringFunction = (value: string) => string;
     type noticeLimitFunction = (maxItemCount: number) => string;
     type callbackOnCreateTemplates = (template: strToEl) => Choices.Templates;
+    type passedElement = HTMLInputElement | HTMLSelectElement;
```
Comment (tinovyatkin) [inline types/index.d.ts]: Most of the changes here are already in `master`, see #682 .

**Comment by jshjohnson** [comment]
@utrumo hey - multiple changes have gone into master, can you verify whether this fix is still needed please?

**Comment by jshjohnson** [comment]
hey @utrumo - I have applied your fix over the latest master branch. Thanks 👍

**Pull request closed without merging**